### ISSUE TITLE
Failure to create S3 bucket when repository name uses mixed case

### ISSUE THREAD

**Issue by cmmeyer** [aws-quickstart/taskcat#155]
**Describe the bug**
**NOTE** Executing via the Quick Start
When using mixed case for my GitHub repo name (CfnLintDemo):
```
·[0;30;43m[INFO ]·[0m :Creating bucket taskcat-tag-CfnLintDemo-0a074b3f in us-east-2
Traceback (most recent call last):
 File "/usr/local/bin/taskcat", line 79, in <module>
 main()
 File "/usr/local/bin/taskcat", line 65, in main
 tcat_instance.stage_in_s3(taskcat_cfg)
 File "/usr/local/lib/python3.6/site-packages/taskcat/stacker.py", line 428, in stage_in_s3
 'LocationConstraint': self.get_default_region()
 File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 320, in _api_call
 return self._make_api_call(operation_name, kwargs)
 File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 623, in _make_api_call
 raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidBucketName) when calling the CreateBucket operation: The specified bucket is not valid.
 [Container] 2018/10/31 04:40:24 Command did not exit successfully taskcat -c $PROJECTNAME/ci/taskcat.yml exit status 1
```

**To Reproduce**
1. Create a GitHub repository with mixed case name (`CfnLintDemo`)
2. Specify repo name as qsname in `taskcat.yml` (`qsname=CfnLintDemo`)
3. Check files into repo.

**NOTE:** Switching the `qsname` to lowercase (`qsname=cfnlintdemo`) yields a different error since it cannot find the repo root directory:

```
·[0;30;43m[INFO ]·[0m :Creating bucket taskcat-tag-cfnlintdemo-69c9ef52 in us-east-2
·[0;30;43m[INFO ]·[0m :Staging Bucket => [taskcat-tag-cfnlintdemo-69c9ef52]
 Hint: The name specfied as value of qsname (cfnlintdemo)
 must match the root directory of your project
·[0;30;41m[ERROR ]·[0m :!Cannot find directory [cfnlintdemo] in /codebuild/output/src391183766
·[0;30;41m[ERROR ]·[0m :Please cd to where you project is located
 [Container] 2018/10/31 04:44:23 Command did not exit successfully taskcat -c $PROJECTNAME/ci/taskcat.yml exit status 1
```

**Expected behavior**
Map qsname to a valid S3 staging bucket.

**Screenshots**

**Version (Please make sure you are running the latest version of taskcat)
 - Taskcat Version version 0.7.19

 Note: Python Version (python3 required)

To find versions:
*Via taskcat*: `taskcat -V`
*Via pip3*: `pip3 show taskcat`

Note: both version should match

To update taskcat run:
 *for docker* : `docker pull taskcat/taskcat`
 *for pip3*: `pip3 install --upgrade taskcat`

**Additional context**

**Comment by agamzn** [comment]
Thanks for the very detailed bug report. This appears to be an S3 naming limitation that's bubbling up. We should convert the bucket-name to lowercase when we create it. 

```
>>> import boto3
>>> session = boto3.session.Session()
>>> client = session.client('s3')
>>>
>>> bucket_name = 'MixedCaseTestBucketFooBar'
>>>
>>> client.create_bucket(Bucket=bucket_name)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    client.create_bucket(Bucket=bucket_name)
  File "/usr/lib/python2.7/dist-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/lib/python2.7/dist-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (InvalidBucketName) when calling the CreateBucket operation: The specified bucket is not valid.
>>>
>>>
>>> client.create_bucket(Bucket=bucket_name.lower())
{u'Location': '/mixedcasetestbucketfoobar', 'ResponseMetadata': {'HTTPStatusCode': 200, 'RetryAttempts': 0, 'HostId': 'kBhMJqVQzGzUxdcj/V+oJLkwWrmZ7tTqrIMzYMB9sQWDdjn9voFs/wDTP3CD31xsFU6bzB/hnNU=', 'RequestId': 'EE666D6FCA15DDBC', 'HTTPHe
aders': {'content-length': '0', 'x-amz-id-2': 'kBhMJqVQzGzUxdcj/V+oJLkwWrmZ7tTqrIMzYMB9sQWDdjn9voFs/wDTP3CD31xsFU6bzB/hnNU=', 'server': 'AmazonS3', 'x-amz-request-id': 'EE666D6FCA15DDBC', 'location': '/mixedcasetestbucketfoobar', 'date':
'Wed, 31 Oct 2018 16:09:54 GMT'}}}
>>>
>>>
```

PR forthcoming.

**Comment by agamzn** [comment]
PR in #157

**Comment by avattathil** [comment]
@agmzn looks good, can you also add a check that ensures that the string length is also compliant when calling bucketcreate

**Comment by agamzn** [comment]
Added a commit. Also addresses #146

**Comment by avattathil** [comment]
Fixed in version `0.7.27`
